lib/Backend/Lower.cpp

```diff
@@ -15167,6 +15169,19 @@ Lowerer::GenerateFastElemICommon(
                 return nullptr;
             }
         }
+        else if (indexOpnd->GetValueType().IsLikelySymbol())
+        {
+            if (!baseOpnd->GetValueType().IsLikelyOptimizedTypedArray())
+            {
+                // If profile data says that it's a typed array - do not generate the symbol fast path as the src. could be a temp and that would cause a bug.
```

Comment:
>could be a temp and that would cause a bug. [](start = 115, length = 43)

I copied this but don't really know if it's necessary in this case or even still relevant. Comment from change that introduced it is below. I removing the typed array checks in the string case above and running typedArrayProfile.js, and I saw no problems.

   Change 1113228 by REDMOND\nikhilkh@IEDRTHOST5-fbliedev1 on 2012/06/16 17:30:09
   
    WIN8:915315: [WebCrawler] [x86chk] [DynamicProfileMutator]:
    INVALID_POINTER_READ jscript9!Js::Type::GetTypeId+c
    [d:\work\enlist\fbl_ie_dev1\src\inetcore\
    The problem is that the PropertyString fast path was getting generated
    within a forin even though the profile data said that the store was on a
    typed array. Using the profile data, the src would be marked temp as it
    was a typed array store. At runtime, the property string fast path was
    actually be successful and store a temp stack allocated javascript
    number.
    When the stack allocated number was used, it would cause an AV.
    I have also added a unit test for this issue.
    IECAT #: 46183
    Code Reviewer: louisl
    Buddy Tester: saagarwa;sansun